lib/rules/media-feature-parentheses-space-inside/index.js

```diff
@@ -31,50 +31,62 @@ const rule = function(expectation) {
       // will be at atRule.raws.params.raw
       const params = _.get(atRule, "raws.params.raw", atRule.params);
       const indexBoost = atRuleParamIndex(atRule);
+      const errors = [];
 
-      styleSearch({ source: params, target: "(" }, match => {
-        const nextCharIsSpace = params[match.startIndex + 1] === " ";
-        if (nextCharIsSpace && expectation === "never") {
-          report({
-            message: messages.rejectedOpening,
-            node: atRule,
-            index: match.startIndex + 1 + indexBoost,
-            result,
-            ruleName
-          });
-        }
-        if (!nextCharIsSpace && expectation === "always") {
-          report({
-            message: messages.expectedOpening,
-            node: atRule,
-            index: match.startIndex + 1 + indexBoost,
-            result,
-            ruleName
-          });
+      const parsedParams = valueParser(params).walk(node => {
+        if (node.type === "function") {
+          const len = valueParser.stringify(node).length;
+          if (expectation === "never") {
+            if (node.before === " ") {
+              if (context.fix) node.before = "";
+              errors.push({
+                message: messages.rejectedOpening,
+                index: node.sourceIndex + 1 + indexBoost
+              });
+            }
+            if (node.after === " ") {
+              if (context.fix) node.after = "";
+              errors.push({
+                message: messages.rejectedClosing,
+                index: node.sourceIndex - 2 + len + indexBoost
+              });
+            }
+          }
+          if (expectation === "always") {
```

Comment:
```suggestion
         else if (expectation === "always") {
```
or 

```suggestion
         else {
```
will be enough due to `validOptions` check on the top of rule definition